src/burnchains/bitcoin/indexer.rs

```diff
@@ -851,6 +851,21 @@ impl BurnchainIndexer for BitcoinIndexer {
             .map_err(burnchain_error::Bitcoin)
     }
 
+    fn get_highest_header_height(&self) -> Result<u64, burnchain_error> {
```

Comment:
What was wrong with `get_headers_height()`?

Reply:
Using `get_headers_height()` was fine (except in one place - burnchain.rs line 508), but introducing `get_highest_header_height` increase legibility: there are many places in the codebase where we're calling `get_header_height()` and subtracting 1 + adding a comment explaining that it's 1-indexed.

Reply:
That's because `get_headers_height()` reports the number of headers the node knows about (i.e. on the longest bitcoin fork).  If you want to convert the number of headers (which is 1-indexed, because there may be 0 headers) into a block height (which is 0-indexed, because block 0 is the first block), then maybe instead of duplicating the `get_headers_height()` implementation, you could add a helper method that converts the number of headers into a block height?  Something like `num_headers_to_block_height(num_headers: u64) -> Option<u64> { num_headers.checked_sub(1) }`.  That would more accurately describe what's going on here.

Reply:
> which is 1-indexed, because there may be 0 headers

This statement and introducing this count as 1-indexed is very confusing, we **always** have at least 1 header in this DB - the genesis header (inserted when initializing these DBs).

Reply:
> Something like num_headers_to_block_height(num_headers: u64) -> Option<u64> { num_headers.checked_sub(1) }. That would more accurately describe what's going on here.

I don't see how 
```
let destination = (source + 1) - 1
```
is more accurately describing
```
let destination = source
```

Reply:
My main concern is that you have two method implementations that do almost the exact same thing.  I think you should either fix `get_headers_height()` so it does what you want (and remove `get_highest_header_height()`), or you should make `get_highest_header_height()` call `get_headers_height()` internally.  If it means changing `get_headers_height()` to be 0-indexed, then that's fine by me.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2073/commits/249aa38a1e3a630f1b5ad6748e25f8f5846a2bc5